### ISSUE TITLE
Public API for mesh axis index size

### ISSUE THREAD

**Issue by gabbard** [jax-ml/jax#25515]
You can use `jax.lax.axis_index(axis)` to get your index along a mesh axis without needing access to the `mesh` object.  However, there does not seem to be a way to get access to the *size* of a mesh axis through the public API.  

We have been doing:
```
        tensor_parallel_size = jax.interpreters.pxla.thread_resources.env.shape["TP"]
```

but `jax.interpreters` [is not part of the public API](https://jax.readthedocs.io/en/latest/api_compatibility.html).

Should such a function be added?  Or are we doing something wrong if we think we sometimes need it?

**Comment by yashk2810** [comment]
No, there is no public API yet but we are working on it.

One thing you can do is create your own context manager and thread the mesh around and access it wherever you want.

**Comment by gabbard** [comment]
Thanks, that's a good idea for a workaround.
